torchx/examples/mnist.exs

```diff
@@ -0,0 +1,187 @@
+defmodule Torchx.MNIST do
+  @batch_size 50
+  @step 0.02
+  @epochs 10
+
+  def init_random_params do
+    w1 = Nx.random_normal({784, 128}, 0.0, 0.1, names: [:input, :layer])
+    b1 = Nx.random_normal({128}, 0.0, 0.1, names: [:layer])
+    w2 = Nx.random_normal({128, 10}, 0.0, 0.1, names: [:layer, :output])
+    b2 = Nx.random_normal({10}, 0.0, 0.1, names: [:output])
+    {w1, b1, w2, b2}
+  end
+
+  defp softmax(logits) do
+    exp = Nx.exp(logits)
+    Nx.divide(exp, Nx.sum(exp, axes: [:output], keep_axes: true))
+  end
+
+  def predict({w1, b1, w2, b2}, batch) do
+    batch
+    |> Nx.dot(w1)
+    |> Nx.add(b1)
+    |> Nx.logistic()
+    |> Nx.dot(w2)
+    |> Nx.add(b2)
+    |> softmax()
+  end
+
+  defp update(
```

Comment:
Btw, this example can still use `defn`! The `defn` evaluator should also default to using the default backend. :) So my suggestion is to keep this as close as possible to the EXLA one!